From 12a46c59ce349a5a61177116cce3d05ccc50d128 Mon Sep 17 00:00:00 2001
From: Jeevaka Badrappan <jeevaka.badrappan@intel.com>
Date: Wed, 24 Apr 2013 09:28:21 +0300
Subject: [PORT FROM MAIN] Telephony: Update WiFi country code

BZ: 57324

Upon use of foreign SIM card, WiFi country
code should be the MCC of the registered network.
Currently, country code of the registered operator
is not updated.

This patch updates the WiFi country code with
the country code received as part of the
registration messages.

Change-Id: Ic480241c26ae2041c71aea0ab86cab76da73fbc0
Orig-Change-Id: I3767f6944398d507e59d1260406cdbb2603b702e
Category: aosp improvement
Domain: Telephony-Telephony-FWK
Origin: internal
Upstream-Candidate: yes
Signed-off-by: Jeevaka Badrappan <jeevaka.badrappan@intel.com>
Signed-off-by: Arun Ravindran <arun.ravindran@intel.com>
---
 .../telephony/gsm/GsmServiceStateTracker.java      |   18 ++++++++++++++++++
 1 files changed, 18 insertions(+), 0 deletions(-)

diff --git a/src/java/com/android/internal/telephony/gsm/GsmServiceStateTracker.java b/src/java/com/android/internal/telephony/gsm/GsmServiceStateTracker.java
index 3d257a7..3e988f8 100644
--- a/src/java/com/android/internal/telephony/gsm/GsmServiceStateTracker.java
+++ b/src/java/com/android/internal/telephony/gsm/GsmServiceStateTracker.java
@@ -27,6 +27,7 @@ import android.content.Intent;
 import android.content.IntentFilter;
 import android.content.res.Resources;
 import android.database.ContentObserver;
+import android.net.wifi.WifiManager;
 import android.os.AsyncResult;
 import android.os.Build;
 import android.os.Handler;
@@ -759,6 +760,21 @@ final class GsmServiceStateTracker extends ServiceStateTracker {
         mSignalStrength = new SignalStrength(true);
     }
 
+    private void setWifiCountryCode(String countryCode) {
+        if (countryCode != null && !countryCode.isEmpty()) {
+            if (DBG) {
+                log("WIFI_COUNTRY_CODE set to " + countryCode);
+            }
+
+            WifiManager wm =
+                    (WifiManager) mPhone.getContext().getSystemService(Context.WIFI_SERVICE);
+            if (wm != null) {
+                //persist
+                wm.setCountryCode(countryCode, true);
+            }
+        }
+    }
+
     /**
      * A complete "service state" from our perspective is
      * composed of a handful of separate requests to the radio.
@@ -965,6 +981,8 @@ final class GsmServiceStateTracker extends ServiceStateTracker {
                 mPhone.setSystemProperty(TelephonyProperties.PROPERTY_OPERATOR_ISO_COUNTRY, iso);
                 mGotCountryCode = true;
 
+                setWifiCountryCode(iso);
+
                 TimeZone zone = null;
 
                 if (!mNitzUpdatedTime && !mcc.equals("000") && !TextUtils.isEmpty(iso) &&
-- 
1.7.4.1

